ui/src/timeMachine/reducers/index.ts

```diff
@@ -649,6 +652,17 @@ export const timeMachineReducer = (
       })
     }
 
+    case 'SET_BUILDER_AGGREGATE_FUNCTION_TYPE': {
+      return produce(state, draftState => {
+        const {index, builderAggregateFunctionType} = action.payload
+        const draftQuery = draftState.draftQueries[draftState.activeQueryIndex]
+
+        draftQuery.builderConfig.tags[
```

Comment:
this should check if index is a valid tag before trying to access it

Reply:
i added the the checks but i'm like 99% certain that the shape of the data in this reducer is consistent with default values